packages/arb-token-bridge-ui/src/util/withdrawals/fetchWithdrawals.ts

```diff
@@ -129,36 +105,33 @@ export const fetchWithdrawals = async ({
   const currentPageStart = pageNumber * pageSize
   const currentPageEnd = currentPageStart + pageSize
 
-  const partialEthWithdrawalsFromEventLogs = [...ethWithdrawalsFromEventLogs]
-    // event logs start from the earliest, we need to reverse them
-    .reverse()
-    .slice(currentPageStart, currentPageEnd)
-  const partialTokenWithdrawalsFromEventLogs = [
-    ...tokenWithdrawalsFromEventLogs
+  // we need timestamps to sort token withdrawals along ETH withdrawals
+  const tokenWithdrawalsFromEventLogsWithTimestamp = await Promise.all(
+    tokenWithdrawalsFromEventLogs.map(withdrawal =>
+      attachTimestampToTokenWithdrawal({ withdrawal, l2Provider })
+    )
+  )
+
+  // get ETH and token withdrawals that will be displayed on the current page
+  const paginatedWithdrawalsFromEventLogs = [
+    ...ethWithdrawalsFromEventLogs,
+    ...tokenWithdrawalsFromEventLogsWithTimestamp
   ]
-    .reverse()
+    .sort((a, b) => (a.timestamp?.gt(b.timestamp || constants.Zero) ? -1 : 1))
     .slice(currentPageStart, currentPageEnd)
 
-  const mappedTokenWithdrawalsFromEventLogs = (
-    await Promise.all(
-      partialTokenWithdrawalsFromEventLogs.map(withdrawal =>
+  const mappedTokenWithdrawalsFromEventLogs = await Promise.all(
+    paginatedWithdrawalsFromEventLogs
+      .filter(withdrawal => !isEthWithdrawal(withdrawal))
```

Comment:
Hm, I don't see why this is necessary, as we should already know if something's an eth withdrawal or a token withdrawal. So we can do the mapping first before we put them all together into a single array and then we don't need this check